### DISCUSSION TITLE
H2 rpm instable

### DISCUSSION THREAD

**Discussion by vmsil** [kachurovskiy/nanoels#127] [Q&A]
Hi Maxim, I'm having issue with counting encoder pulses.
RPM reading is slightly changing (approx. 300/-+8rpm) and this ruins my thread.
This probably is caused by interference in cables or something like that, I need to check the signal with a scope when I get one.
What I'm wondering is that I'm using LCD I2C display and on some thread you mentioned
that it is slower to update display with I2C, can this cause issue for me?
My coding skills are so minimal that I cannot see from the code how display update is being handled :)
My build is H2 sw7.

**Comment by kachurovskiy** [comment]
Hi @vmsil,

1. Try setting `DIR_CHANGE_DIFF_MICROS` to 0 in https://github.com/kachurovskiy/nanoels/blob/main/h2/h2.ino#L51C9-L51C32 and let us know if it helps.

2. Can you also please send a picture of your encoder installation and how the wires are coming into the controller?

3. Try making a mark on the spindle, turning on Angle indication on the display and seeing if turning the chuck by hand results in Angle returning to the same value. Are pulses lost or are extra pulses counted?

> slower to update display with I2C, can this cause issue for me?

That is unlikely, I2C is ~20% slower if I remember right but it should not result in lost pulses.

Thanks!
